### PR TITLE
First v1 -> v2 transformer (minimum log)

### PR DESCRIPTION
**Pull request by EasyRhin0**

### CODE REVIEW

**Comment by lgolding** [comment]
File src/Sarif.FunctionalTests/ConverterTestData/PREfast/src/C6001/.vscode/.browse.VC.db shouldn't be there. Add .browse.VC.db to .gitignore.

**Comment by EasyRhin0** [comment]
Part of PREfast, it’s there in master.

**Comment by lgolding** [comment]
@EasyRhin0 Then it's a mistake in master, too. Kill it with fire.

File: src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs [hunk 1]
```diff
@@ -1,20 +1,86 @@
...
+        public override RunVersionOne VisitRunVersionOne(RunVersionOne node)
+        {
+            if (node != null)
+            {
+                Run run = new Run()
+                {
+                    Architecture = node.Architecture,
+                    BaselineId = node.BaselineId,
+                    AutomationId = node.AutomationId,
+                    DefaultFileEncoding = "???",
```
Comment (EasyRhin0) [inline src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs]: ??? Just placeholders, assuming these properties will be default

**Comment by lgolding** [comment]
This file is still there; it's a casing issue in .gitignore:

src/Sarif.FunctionalTests/ConverterTestData/PREfast/src/C6001/.vscode/.browse.VC.db

**Comment by EasyRhin0** [comment]
It’s case sensitive?

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using FluentAssertions;
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: Copyright notice on new files (see existing files for example).

**Comment by lgolding** [comment]
@EasyRhin0 Actually I don't know. But come to think of it, since Git is _already_ tracking that file, you have to "git rm" it.

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+            v2Log.Runs[0].Results.Should().NotBeNull();
+            v2Log.Runs[0].Results.Count.Should().Be(0);
+
+            v2Log.Runs[0].Tool.Should().NotBeNull();
+            v2Log.Runs[0].Tool.Name.Should().Be("CodeScanner");
+            v2Log.Runs[0].Tool.SemanticVersion.Should().Be("2.1.0");
+        }
+    }
+}
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: _Nice_ first test. Now we're up and running!

File: src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs (hunk 1 above)
Comment (lgolding) [reply src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs]: I don't understand this comment. Also, if there is something to be explained here, let's put the comment in the code.

Comment (EasyRhin0) [reply src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs]: Just noting that I'm not going to leave it that way. I will push new changes.

File: src/Sarif.UnitTests/SarifTransformerTests.cs (hunk 2 above)
Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifTransformerTests.cs]: Fixed

File: src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs [hunk 4]
```diff
@@ -1,20 +1,86 @@
...
+                };
+
+                foreach (string propertyName in node.PropertyNames)
+                {
+                    tool.PropertyNames.Add(propertyName);
+                }
+
+                tool.Tags.UnionWith(node.Tags);
+
+                SarifLog.Runs.Last().Tool = tool;
```
Comment (lgolding) [inline src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs]: Huh. I guess this _will_ work for multiple runs. Let's add a test for that.

**Pull request merged**